cartoframes/context.py

```diff
@@ -234,16 +233,89 @@ def read(self, table_name, limit=None, index='cartodb_id',
         # choose schema (default user - org or standalone - or shared)
         schema = 'public' if not self.is_org else (
             shared_user or self.creds.username())
-        query = 'SELECT * FROM "{schema}"."{table_name}"'.format(
+
+        table_columns = self._get_table_columns(table_name, schema)
+
+        query = self._get_read_query(table_name, schema, table_columns, limit)
+        result = self.copy_client.copyto_stream(query)
+        df = pd.read_csv(result)
+
+        return self._clean_dataframe_from_carto(df, table_columns, decode_geom)
+
+    def _get_read_query(self, table_name, schema, table_columns, limit=None):
```

Comment:
I don't see a reason for this to be a class member

Reply:
Same as the following one